src/robotFactory.js

```diff
@@ -1,10 +1,81 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, { x = 0, y = 0 }, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x: x,
+      y: y,
+    };
```

Comment:
If you have same key/value you can write it once
```suggestion
    this.coords = { x, y };
```